### PR TITLE
Dev ag inscription ti back

### PR DESCRIPTION
**Pull request by gonzalezad1**

### CODE REVIEW

File: db/migrations/20181015112039_users_tis.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            .inTable("users");
+        table.string("email");
+        table.string("cabinet");
+        table.dateTime("creates_at").defaultTo(knex.fn.now());
+    });
+};
+
+exports.down = function(knex, Promise) {
+    return knex.schema.dropTable("users_tis");
+};
```
Comment (douglasduteil) [inline db/migrations/20181015112039_users_tis.js]: missing EOL

File: db/migrations/20181019112457_usersMandataires.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      .integer("user_id")
+      .references("id")
+      .inTable("users");
+    table.dateTime("creates_at").defaultTo(knex.fn.now());
+  });
+};
+
+exports.down = function(knex, Promise) {
+  return knex.schema.dropTable("users_mandataires");
+};
```
Comment (douglasduteil) [inline db/migrations/20181019112457_usersMandataires.js]: Missing EOL

File: db/queries/mesures.js [hunk 3]
```diff
@@ -91,7 +106,7 @@ const getAllMesuresByMandatairesFilter = (
         .where("users.type", "service")
         .where("mandataire_tis.ti_id", parseInt(ti_id));
     });
-
+//TODO : see where it was used and rm if necessary
```
Comment (douglasduteil) [inline db/queries/mesures.js]: Prefer adding your github name here `TODO(<name>): <message>`

File: routes/admin.js [hunk 4]
```diff
@@ -109,6 +109,34 @@ router.get("/mandataires", typeRequired("admin"), async (req, res, next) => {
...
+ *         description: filters to apply to the list. ex `?users.active=true`
+ *         required: false
+ *         schema:
+ *           type: object
+ *     responses:
+ *       200:
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
```
Comment (douglasduteil) [inline routes/admin.js]: `array` of what `User` no ?

File: routes/admin.js [hunk 5]
```diff
@@ -109,6 +109,34 @@ router.get("/mandataires", typeRequired("admin"), async (req, res, next) => {
...
+ *   get:
+ *     tags:
+ *       - admin
+ *     description: get list of users of Ti for admin
+ *     produces:
+ *       - application/json
+ *     parameters:
+ *       - in: query
+ *         name: query
+ *         description: filters to apply to the list. ex `?users.active=true`
```
Comment (douglasduteil) [inline routes/admin.js]: The name is `query` no ? So it should be `/admin/tis?query=<sth>` no?

File: routes/inscription.js [hunk 6]
```diff
@@ -221,4 +257,71 @@ router.post("/mandataires", (req, res, next) => {
...
+                  email
+                },
+                trx
+              )
+            )
+          );
+        })
+    )
+    .then(() => {
+      // todo: send email admins ?
```
Comment (douglasduteil) [inline routes/inscription.js]: Prefer adding your github name here `TODO(<name>): <message>`

File: routes/mesures.js [hunk 7]
```diff
@@ -132,6 +132,8 @@ router.post("/filters", loginRequired, async (req, res, next) => {
  */
 router.get("/popup", typeRequired("ti"), async (req, res, next) => {
   const ti = await getTiByUserId(req.user.id);
+  console.log("getAllMesuresByTis", ti)
```
Comment (douglasduteil) [inline routes/mesures.js]: For debugging only no ?

File: scripts/migrateMandataireToUsersMandataire.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+const {
+  Mandataires,
+} = require("../db/queries/mandataires");
+
+const knex = require("../db/knex.js");
+
+const migrationMandataireUserIdTOMandatairesUsersID = () => {
+
+
```
Comment (douglasduteil) [inline scripts/migrateMandataireToUsersMandataire.js]: Are you using thie file ? :thinking:

File: test/routes.inscription.test.js [hunk 9]
```diff
@@ -151,6 +151,35 @@ describe("routes : inscription", () => {
...
+          pass1: "kikoo",
+          pass2: "kikoo",
+          cabinet: "2A",
+          tis: [1]
+        })
+        .then((res, req) => {
+          res.status.should.eql(200);
+        })
+        .then(async (res, req) => {
+          const user = await knex
```
Comment (douglasduteil) [inline test/routes.inscription.test.js]: Mix of async/await and `.then` function call back. 
You should make the whole `it` `async` :heart:

File: test/routes.admin.test.js [hunk 10]
```diff
@@ -80,6 +80,36 @@ describe("routes : /admin", () => {
...
+      logUser(server, {
+        username: "admin",
+        password: "admin"
+      }).then(token =>
+        agent
+          .get("/api/v1/admin/tis")
+          .set("Authorization", "Bearer " + token)
+          .then(res => {
+            res.status.should.equal(200);
+            res.body.length.should.equal(2);
```
Comment (revolunet) [inline test/routes.admin.test.js]: please check the content too

File: test/routes.admin.test.js [hunk 11]
```diff
@@ -80,6 +80,36 @@ describe("routes : /admin", () => {
...
+      logUser(server, {
+        username: "admin",
+        password: "admin"
+      }).then(token =>
+        agent
+          .get("/api/v1/admin/tis?users.active=false")
+          .set("Authorization", "Bearer " + token)
+          .then(res => {
+            res.status.should.equal(200);
+            res.body.length.should.equal(1);
```
Comment (revolunet) [inline test/routes.admin.test.js]: please check the content too

File: routes/inscription.js [hunk 12]
```diff
@@ -224,4 +259,73 @@ router.post("/mandataires", (req, res, next) => {
...
+                  email
+                },
+                trx
+              )
+            )
+          );
+        })
+    )
+    .then(() => {
+      return inscriptionEmail(`${process.env.APP_URL}/admin`);
```
Comment (revolunet) [inline routes/inscription.js]: remove "admin" ?

**Pull request merged**